sql/src/main/java/org/apache/druid/sql/SqlLifecycle.java

```diff
@@ -131,12 +139,28 @@ private String sqlQueryId()
     return (String) this.queryContext.get(PlannerContext.CTX_SQL_QUERY_ID);
   }
 
+  public void setParameters(List<TypedValue> parameters)
+  {
+    this.parameters = parameters;
+  }
+
+  public PrepareResult prepare(AuthenticationResult authenticationResult) throws ValidationException, RelConversionException, SqlParseException
+  {
+    synchronized (lock) {
+      try (DruidPlanner planner = plannerFactory.createPlanner(queryContext, parameters, authenticationResult)) {
+        this.plannerContext = planner.getPlannerContext();
```

Comment:
It doesn't _need_ to be, but since logging/emitting metrics collects information from it, if something goes wrong before it is set it makes the data available for that part.